src/Services/Air/Air.js

```diff
@@ -283,54 +283,66 @@ module.exports = (settings) => {
     },
 
     cancelPNR(options) {
-      return this.getUniversalRecordByPNR(options).then(record => this.getTickets({
-        reservationLocatorCode: record.uapi_reservation_locator,
-      })
+      return this.getUniversalRecordByPNR(options)
         .then(
-          tickets => Promise.all(tickets.map(
-            (ticketData) => {
-              // Check for VOID or REFUND
-              const allTicketsVoidOrRefund = ticketData.tickets.every(
-                ticket => ticket.coupons.every(
-                  coupon => coupon.status === 'V' || coupon.status === 'R'
-                )
-              );
-              if (allTicketsVoidOrRefund) {
-                return Promise.resolve(true);
-              }
-              // Check for cancelTicket option
-              if (options.cancelTickets !== true) {
-                return Promise.reject(new AirRuntimeError.PNRHasOpenTickets());
-              }
-              // Check for not OPEN/VOID segments
-              const hasNotOpenSegment = ticketData.tickets.some(
-                ticket => ticket.coupons.some(
-                  coupon => 'OV'.indexOf(coupon.status) === -1
-                )
-              );
-              if (hasNotOpenSegment) {
-                return Promise.reject(new AirRuntimeError.UnableToCancelTicketStatusNotOpen());
-              }
-              return Promise.all(
-                ticketData.tickets.map(
-                  ticket => (
-                    ticket.coupons[0].status !== 'V'
-                      ? service.cancelTicket({
-                        pnr: options.pnr,
-                        ticketNumber: ticket.ticketNumber,
-                      })
-                      : Promise.resolve(true)
-                  )
-                )
-              );
-            }
-          ))
+          ur => getBookingFromUr(ur, options.pnr)
```

Comment:
I recommend you to use `this.getTickets(options)` instead, as tickets from import call could be corrupted. That's why tickets are filtered in response.